### PR TITLE
Bumps sso gem

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

HOTT-<TODO>

### What?

I have added/removed/altered:

- [x] Bumps gds-sso gem to 15.0.1

### Why?

I am doing this because:

- This is needed to keep us up-to-date with the latest patches from the client and signon service

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -10,6 +10,7 @@ ruby '~> 2.7.1'
 # Server
 gem 'puma', '~> 5.0.4'
 gem 'rails', '>= 6.0.3.4'
+gem 'mimemagic'
```
Comment (willfish) [inline Gemfile:13]: This is now completely removed as a dependency from this project. I've added it here to reflect that we still use the mimetype functionality of the mimemagic gem

File: Gemfile.lock [hunk 2]
```diff
@@ -1,61 +1,65 @@
 GEM
   remote: https://rubygems.org/
   specs:
-    actioncable (6.0.3.4)
-      actionpack (= 6.0.3.4)
+    actioncable (6.1.3.1)
```
Comment (willfish) [inline Gemfile.lock:4]: These get updated when you unlock the dependency graph and update the gds-sso gem

File: app/models/user.rb [hunk 3]
```diff
@@ -50,6 +50,12 @@ def disabled?
     disabled
   end
 
+  def update!(options)
```
Comment (willfish) [inline app/models/user.rb:53]: Add a test of me

Comment (willfish) [reply app/models/user.rb:53]: This is actually tested by the shared examples

**Review by mafonso — APPROVED** [review]
Pair reviewed.

LGTM 

:shipit:

**Pull request merged**